yt_dlp/extractor/academymel.py

```diff
@@ -0,0 +1,83 @@
+
+    _TESTS = [{
+        'url': 'http://academymel.online/3video_1',
+        'info_dict': {
+            'id': '4885302',
+            'title': 'Промоуроки Академии МЕЛ',
+            'ext': 'mp4',
+            'duration': 1693
+        }
+    }]
+
+    def _perform_login(self, username, password):
+        login_body = urlencode_postdata({
+            'action': 'processXdget',
+            'xdgetId': 'r6335_1_1',
+            'params[action]': 'login',
+            'params[url]': 'http://academymel.online/cms/system/login?required=true',
+            'params[object_type]': 'cms_page',
+            'params[object_id]': -1,
+            'params[email]': username,
+            'params[password]': password,
+            'requestTime': int(time.time())
+        })
+
+        self._request_webpage(self._LOGIN_URL,
+                              None,
+                              data=login_body,
+                              note='Logging into the academymel.online',
+                              errnote='Failed to log in into academymel.online',
+                              fatal=True)
+
+    def playlist_from_entries(self, entries, valid_url):
+        current_timestamp = int(time.time())
+        current_datetime = datetime.fromtimestamp(current_timestamp)
+        formatted_datetime = current_datetime.strftime("%d.%m.%Y, %H:%M")
+
+        return self.playlist_result(entries,
+                                    'academymel-playlist-%d' % current_timestamp,
+                                    'AcademyMel playlist (%s)' % formatted_datetime,
+                                    'AcademyMel playlist for %s (at %s)' % (valid_url, formatted_datetime))
+
+    def _real_extract(self, url):
+        valid_url = self._match_valid_url(url)
+
+        if not valid_url:
+            raise ExtractorError('Invalid URL found', expected=True)
+
+        webpage = self._download_webpage(url,
+                                         None,
+                                         fatal=True,
+                                         note='Downloading video website',
+                                         errnote='Failed to download video website')
+
+        title = self._search_regex(r'<title>(?P<title>.*)</title>', webpage, 'title')
+
+        entries = []
+        processed_urls = set()  # Set to keep track of processed URLs
+
+        for video_url in re.findall(
+            r'data-iframe-src=\"(?P<url>https?://[^/]+\.getcourse\.ru/sign-player/\?.*?)\"',
+            webpage,
+                re.DOTALL + re.VERBOSE):
+            # Check if the URL has not been processed before
+            if video_url not in processed_urls:
+                entries.append(self.url_result(video_url, 'GetCourseRu', url_transparent=True, title=title))
+                processed_urls.add(video_url)  # Add the URL to the set of processed URLs
+
+        return self.playlist_from_entries(entries, valid_url)
```

Comment:
this extractor is an ideal candidate for `playlist_from_matches`, which uses `yt_dlp.utils.orderedSet` to de-dupe the url results
```suggestion
        return self.playlist_from_matches(
            re.findall(r'data-iframe-src="(https?://[^."]+\.getcourse\.ru/sign-player/[^"]+', webpage),
            playlist_id, title, ie=GetCourseRuIE, video_kwargs={
                'url_transparent': True,
                'title': title,
            })
```